core/azure-core/src/main/java/com/azure/core/amqp/exception/ErrorContext.java

```diff
@@ -17,18 +17,24 @@
     private static final long serialVersionUID = -2819764407122954922L;
 
     private final String namespaceName;
+    private final Throwable exception;
 
     /**
      * Creates a new instance with the provided {@code namespaceName}.
      *
-     * @param namespaceName Namespace of the error context.
+     * @param exception Exception that caused this error.
+     * @param namespaceName Event Hub namespace of the error context.
+     * @throws IllegalArgumentException when 'namespaceName' is {@code null} or empty.
      */
-    public ErrorContext(final String namespaceName) {
+    public ErrorContext(final Throwable exception, final String namespaceName) {
+        Objects.requireNonNull(exception);
```

Comment:
Since you're covering this case below, should we consider removing this so that the message is consistent for `null` and `""` ?

Reply:
It will throw a NullPointerException if exception == null. I updated the docs to state this.